### PR TITLE
[WinUI] Fix the size returned when measuring Frame

### LINKED ISSUE #5600: Frame does not resize
### Description

When resizing the main window with root content `Frame` it doesn't get resized.
The provided xaml code works in **Xamarin Forms**
xamarinforms_frame_resize

Behavior on **dotnet maui** 
maui_frame_doesnot_resize

### Steps to Reproduce

1. Create an ew Maui Project. 
2. Add as content in `MainPage.xaml`
3. 
```xaml
<?xml version="1.0" encoding="utf-8" ?>
<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             x:Class="MauiApp3.MainPage"
             BackgroundColor="{DynamicResource SecondaryColor}">

    <Frame CornerRadius="30" 
           HasShadow="True"
           IsClippedToBounds="True"
           Padding="0">
        <Grid RowSpacing="25" RowDefinitions="Auto,Auto,Auto,Auto,*" Margin="0"
              Padding="{OnPlatform iOS='30,60,30,30', Default='30'}" Background="Red">

            <Label 
                Text="Hello, World!"
                Grid.Row="0"
    
                FontSize="32"
                HorizontalOptions="Center" />

            <Label 
                Text="Welcome to .NET Multi-platform App UI"
                Grid.Row="1"

                FontSize="18"
                HorizontalOptions="Center" />

            <Label 
                Text="Current count: 0"
                Grid.Row="2"
                FontSize="18"
                FontAttributes="Bold"
                x:Name="CounterLabel"
                HorizontalOptions="Center" />

            <Button 
                Text="Click me"
                FontAttributes="Bold"
                Grid.Row="3"
   

                HorizontalOptions="Center" />

            <Image Grid.Row="4"
                Source="dotnet_bot.png"
    
                WidthRequest="250"
                HeightRequest="310"
                HorizontalOptions="Center" />

        </Grid>
    </Frame>
</ContentPage>
```

### Version with bug

Preview 14 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

net6.0-windows10.0.19041, 10.0.17763.0

### Did you find any workaround?

Switch to XF

### Relevant log output

_No response_

### LINKED ISSUE #9937: [WinUI] Frame with an Entry causes app to freeze
### Description

Whenever an Entry underneath a Frame has its properties Text and IsPassword set, the app freezes when ran on WinUI. I've attached a sample app that showcases the issue.

### Steps to Reproduce

1. Open the MauiApp1 app I've attached
2. Run the app on WinUI
3. App freezes
[MauiApp1.zip](https://github.com/dotnet/maui/files/9495008/MauiApp1.zip)

Without opening the sample app:
1. Create a new Maui app
2. Add a StackLayout to the MainPage.xaml
3. Add a Frame to the StackLayout
4. Add an Entry to the Frame
5. Set the Text property of the Entry to for instance "blabla"
6. Set the IsPassword property of the Entry to True
7. Run the app on WinUI
8. The app freezes

### Link to public reproduction project repository

https://github.com/ProgressiveDeyo/MauiApp1

### Version with bug

6.0.486 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

Windows SDK 10

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### LINKED ISSUE #10668: Basic Frame Test no longer passes on WinUI
### Description

- On the `BasicFrameTest` first change this line
```C#
frame.ToHandler(MauiContext).PlatformView.AttachAndRun(async () =>
```
to
```C#
frame.ToPlatform(MauiContext).AttachAndRun(async () =>
```
- Run Test and see that it fails.

This test is just validating that you can center inside a frame for WinU.

### LINKED ISSUE #11990: Frame does not render border correctly while corner radius specified on windows platform
### Description

We are using the frame control for showing control with corner radius and border, but the border was not render correctly while corner radius specified on windows platform.

```
 <Grid HeightRequest="400" WidthRequest="350" >
            <Frame Background="White" BorderColor="#E6E6E6" Padding="0" Margin="0">
                <Grid Background="Green"/>
            </Frame>
</Grid>
```

Issue

**Issue:** 
Issue

**Note:** We used border control to replace the frame control but border control blur its child(text drawn on drawable view) on windows.

### Steps to Reproduce

1. Create the application with above specified code
2. Run the application 
3. Output UI shown some glitch on left top, right top, left bottom and right bottom of frame

### Link to public reproduction project repository

Not Applicable

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

Windows 10.0.17134

### Did you find any workaround?

No

### Relevant log output

```shell
Not applicable
```

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change

The `Frame` and `VER` code was measuring the `Content` of the `Frame` with the overall available size and then returning that for the measure result. This caused the `Frame` to ignore any sizes that were being set on the `Frame` itself.  This PR fixes it so that the measure size of the `Frame` gets returned from `MeasureOverride` and those constraints are used to measure the children as well. 

I grabbed the `XAML` from this issue as well to verify and now the two `Frames` render
https://github.com/dotnet/maui/pull/10964#issuecomment-1329205563
image

This PR also opts `FrameRenderer` into all the same handler tests we have inside core. 

### Issues Fixed

Fixes #5600
Fixes #10668
Fixes #9937
Fixes #11990

### CODE REVIEW

File: src/Controls/src/Core/Compatibility/Handlers/Windows/VisualElementRenderer.cs [hunk 1]
```diff
@@ -89,9 +89,7 @@ protected override WSize MeasureOverride(global::Windows.Foundation.Size availab
 			{
 				if (child is Maui.IElement childElement && childElement.Handler is IPlatformViewHandler nvh)
 				{
-					var size = nvh.GetDesiredSizeFromHandler(availableSize.Width, availableSize.Height);
-					height = Math.Max(height, size.Height);
-					width = Math.Max(width, size.Width);
+					_ = nvh.GetDesiredSizeFromHandler(width, height);
```
Comment (PureWeen) [inline src/Controls/src/Core/Compatibility/Handlers/Windows/VisualElementRenderer.cs:92]: This change only effects `FrameRenderer`. None of our other classes (ListView, TableView) that inherit from `VisualElementRenderer` have `LogicalChildren`

File: src/Controls/src/Core/Compatibility/Handlers/Android/FrameRenderer.cs [hunk 2]
```diff
@@ -50,9 +50,19 @@ public class FrameRenderer : CardView, IPlatformViewHandler
...
 		public event EventHandler<PropertyChangedEventArgs>? ElementPropertyChanged;
 
-		public FrameRenderer(Context context) : base(context)
+		public FrameRenderer(Context context) : this(context, Mapper)
 		{
-			_viewHandlerWrapper = new ViewHandlerDelegator<Frame>(Mapper, CommandMapper, this);
+		}
+
+		// TODO NET8 make public
+		internal FrameRenderer(Context context, IPropertyMapper mapper)
```
Comment (Redth) [inline src/Controls/src/Core/Compatibility/Handlers/Android/FrameRenderer.cs:58]: Should our process here be to start these off as public in the PR to main and then when we backport, change to internal?

File: src/Controls/src/Core/Compatibility/Handlers/Windows/VisualElementRenderer.cs (hunk 1 above)
Comment (Redth) [reply src/Controls/src/Core/Compatibility/Handlers/Windows/VisualElementRenderer.cs:92]: Can you add this as a comment in the code too?

Comment (PureWeen) [reply src/Controls/src/Core/Compatibility/Handlers/Windows/VisualElementRenderer.cs]: I backed these changes out of this PR. The PR now only has changes inside `FrameRenderer`

File: src/Controls/src/Core/Compatibility/Handlers/Windows/FrameRenderer.cs [hunk 3]
```diff
@@ -13,12 +13,33 @@ namespace Microsoft.Maui.Controls.Handlers.Compatibility
 	public class FrameRenderer : ViewRenderer<Frame, WBorder>
 	{
 		public static IPropertyMapper<Frame, FrameRenderer> Mapper
-			= new PropertyMapper<Frame, FrameRenderer>(VisualElementRendererMapper);
+			= new PropertyMapper<Frame, FrameRenderer>(VisualElementRendererMapper)
+			{
+				// https://github.com/dotnet/maui/issues/11880
```
Comment (PureWeen) [inline src/Controls/src/Core/Compatibility/Handlers/Windows/FrameRenderer.cs:18]: Fixing the above issue would most likely make adding these mappers unnecessary. If said fix could be back ported to .NET7 then 👍but if not, then these changes are still relevant. 

I tried to focus these changes purely around `Frame` to make this more back portable.  I'm not aware of other open issues that would be resolved by fixing #111880 currently which is another reason I'm going with the current assumption that we wouldn't fix #11880 for .NET 7,

**Review by hartez — APPROVED** [review]

**Pull request merged**